airbyte-integrations/connectors/source-mysql/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlSelectQuerier.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.mysql
+
+import io.airbyte.cdk.jdbc.JdbcConnectionFactory
+import io.airbyte.cdk.read.JdbcSelectQuerier
+import io.airbyte.cdk.read.SelectQuerier
+import io.airbyte.cdk.read.SelectQuery
+import io.github.oshai.kotlinlogging.KotlinLogging
+import io.micronaut.context.annotation.Primary
+import jakarta.inject.Singleton
+
+private val log = KotlinLogging.logger {}
+
+/** Mysql implementation of [JdbcSelectQuerier], which sets fetch size differently */
+@Singleton
+@Primary
+class MysqlSelectQuerier(
+    jdbcConnectionFactory: JdbcConnectionFactory,
+) : JdbcSelectQuerier(jdbcConnectionFactory) {
+    override fun executeQuery(
+        q: SelectQuery,
+        parameters: SelectQuerier.Parameters,
+    ): Result = MysqlResult(jdbcConnectionFactory, q, parameters)
+
+    inner class MysqlResult(
+        jdbcConnectionFactory: JdbcConnectionFactory,
+        q: SelectQuery,
+        parameters: SelectQuerier.Parameters,
+    ) : JdbcSelectQuerier.Result(jdbcConnectionFactory, q, parameters) {
+        /**
+         * Mysql does things differently with fetch size. Setting fetch size on a result set is
+         * safer than on a statement.
+         */
+        override fun initQueryExecution() {
+            conn = jdbcConnectionFactory.get()
+            stmt = conn!!.prepareStatement(q.sql)
+            stmt!!.fetchSize = Int.MIN_VALUE
```

Comment:
Note: in mysql jdbc driver int.min indicates streaming of records one by one. 
This proved to be safer way of returning results.
After the first record arrived, we set a fetch size on the resultset.